core/src/integration-test/java/com/scalar/db/api/DistributedStorageAdminRepairTableIntegrationTestBase.java

```diff
@@ -0,0 +1,139 @@
+
+  private static final String TABLE = "test_table";
+  private static final String COL_NAME1 = "c1";
+  private static final String COL_NAME2 = "c2";
+  private static final String COL_NAME3 = "c3";
+  private static final String COL_NAME4 = "c4";
+  private static final String COL_NAME5 = "c5";
+  private static final String COL_NAME6 = "c6";
+  private static final String COL_NAME7 = "c7";
+  private static final String COL_NAME8 = "c8";
+  private static final String COL_NAME9 = "c9";
+  private static final String COL_NAME10 = "c10";
+  private static final String COL_NAME11 = "c11";
+
+  protected static final TableMetadata TABLE_METADATA =
+      TableMetadata.newBuilder()
+          .addColumn(COL_NAME1, DataType.INT)
+          .addColumn(COL_NAME2, DataType.TEXT)
+          .addColumn(COL_NAME3, DataType.TEXT)
+          .addColumn(COL_NAME4, DataType.INT)
+          .addColumn(COL_NAME5, DataType.INT)
+          .addColumn(COL_NAME6, DataType.TEXT)
+          .addColumn(COL_NAME7, DataType.BIGINT)
+          .addColumn(COL_NAME8, DataType.FLOAT)
+          .addColumn(COL_NAME9, DataType.DOUBLE)
+          .addColumn(COL_NAME10, DataType.BOOLEAN)
+          .addColumn(COL_NAME11, DataType.BLOB)
+          .addPartitionKey(COL_NAME2)
+          .addPartitionKey(COL_NAME1)
+          .addClusteringKey(COL_NAME4, Scan.Ordering.Order.ASC)
+          .addClusteringKey(COL_NAME3, Scan.Ordering.Order.DESC)
+          .addSecondaryIndex(COL_NAME5)
+          .addSecondaryIndex(COL_NAME6)
+          .build();
+
+  protected DistributedStorageAdmin admin;
+
+  protected AdminTestUtils adminTestUtils;
+
+  protected void initialize() throws Exception {}
+
+  protected abstract Properties getProperties();
+
+  protected Properties getStorageProperties() {
+    return getProperties();
+  }
+
+  protected String getNamespace() {
+    return NAMESPACE;
+  }
+
+  protected String getTable() {
+    return TABLE;
+  }
+
+  private void createTable() throws ExecutionException {
+    Map<String, String> options = getCreateOptions();
+    admin.createNamespace(getNamespace(), options);
+    admin.createTable(getNamespace(), getTable(), TABLE_METADATA, options);
+  }
+
+  protected Map<String, String> getCreateOptions() {
+    return Collections.emptyMap();
+  }
+
+  private void dropTable() throws ExecutionException {
+    admin.dropTable(getNamespace(), getTable());
+    admin.dropNamespace(getNamespace());
+  }
+
+  @BeforeAll
+  protected void beforeAll() throws Exception {}
+
+  @BeforeEach
+  protected void beforeEach() throws Exception {
+    StorageFactory factory = StorageFactory.create(TestUtils.addSuffix(getProperties(), TEST_NAME));
+    admin = factory.getStorageAdmin();
+    createTable();
+    adminTestUtils = AdminTestUtils.create(TestUtils.addSuffix(getStorageProperties(), TEST_NAME));
+  }
+
+  @AfterEach
+  protected void afterEach() throws Exception {
+    dropTable();
+  }
+
+  @AfterAll
+  protected void afterAll() throws Exception {}
```

Comment:
We should close `admin` just in case.
```suggestion
  protected void afterAll() throws Exception {
    admin.close();
  }
```